cluster/gce/gci/configure-helper.sh

```diff
@@ -1858,6 +1858,18 @@ function prepare-etcd-manifest {
   fi
   # Replace the volume host path.
   sed -i -e "s@/mnt/master-pd/var/etcd@/mnt/disks/master-pd/var/etcd@g" "${temp_file}"
+  # Replace the runAsUser and runAsGroup
+  sed -i -e "s@{{runAsUser}}@${ETCD_RUNASUSER:-0}@g" "${temp_file}"
```

Comment:
I think it would be safest to not edit the manifest if ETCD_RUNASUSER and ETCD_RUNASGROUP is not set. Basically if ETCD_RUNASUSER and ETCD_RUNASGROUP is not set then this change is a noop, WDYT?

Reply:
We have to replace the {{runAsUser}} and {{runAsGroup}} in the manifest anyway isn't? I mean replace it to 0 if ETCD_RUNASUSER and ETCD_RUNASGROUP is not set.

Reply:
Why do that? If we don't set it then it will anyways run as 0.

Reply:
The manifest is written as:
```
"runAsUser": {{runAsUser}},
"runAsGroup": {{runAsGroup}}
```
The {{runAsUser}} and group need to be replace at least with some value, or are we talking about different thing?

Reply:
No I am talking about this, but I think we can make the manifest to be:-
```
 securityContext {
 ...
 {{runAsUser}}
 {{runAsGroup}}
 }
```
and replace `{{runAsUser}}` with `"\"runAsUser\": ${ETCD_RUNASUSER}"` and same for runAsGroup. That way we only insert it ETCD_RUNASUSER else we replace {{runAsUser}} with empty string.

Reply:
Do we want to replace the entire securityContext as empty when it's root? Otherwise when it's root and runAsUser and runAsGroup is empty, the delimiter  `,` at  `seccompProfile` will  be a problem.
```
"securityContext": {
    "seccompProfile": {
        "type": "RuntimeDefault"
    },
    "runAsUser": {{runAsUser}},
    "runAsGroup": {{runAsGroup}}
}
```

Reply:
I really think we should fix this.

Reply:
fixed.